File: C Implementation/Queue.cpp
#include<stdio.h>
#include<stdlib.h>
#define MAX 10
int arr[MAX];
int Front=-1,Rear=-1;

bool IsEmpty();
bool IsFull();
void push(int val);
void pop();
void print();

bool IsEmpty()
{
    if(Front==-1 && Rear==-1)
    {
        return true;
    }
    else
        return false;
}

bool IsFull()
{
    if((Rear+1)%MAX == Front)
    {
        return true;
    }
    else
        return false;
}

void print()
{
    if(IsEmpty()==true)
    {
        printf("No value inside QUEUE\n");
        return;
    }
    else
    {
        printf("\nQueue is: ");
        for(int i=Front; i<=Rear; i++)
        {
            printf("%d ",arr[i]);
        }
    }
}

void push(int val)
{
    if(IsFull()==true)
    {
        printf("Queue is Full!!");
        return;
    }
    else if(IsEmpty())
    {
        Front=Rear=0;
    }
    else
    {
        Rear+=1;
    }
    arr[Rear]=val;
}

void pop()
{
    if(IsEmpty()==true)
    {
        printf("Queue is empty!!");
        return;
    }
    else if(Front==Rear)
    {
        Front=Rear=-1;
    }
    else
    {
      // Front = (Front+1)%MAX;
       Front +=1;
    }
}
void Selection(int ch)
{
 int i,val;
    switch(ch)
    {
    case 1:
        printf("\nPlease Enter a value:");
        scanf("%d",&val);
        push(val);
        print();
        break;
    case 2:
        pop();
        print();
        break;
    default:
        printf("please Enter a correct  value");
        break;
    }

}

int main()
{
    int ch;
    while(true)
    {
        printf("1: To Push a value inside Queue\n2: To pop a value from QUEUE\n");
        scanf("%d",&ch);
        Selection(ch);
        printf("\n");
    }

    return 0;
}

File: Java Implementation/Queue/src/queue/Queue.java

package queue;

import java.util.Scanner;

class Queue_Operate
{
    int Front=-1;
    int Rear=-1;
    final int MAX=5;
    int[] queue = new int[MAX];
    
    boolean isEmpty()
    {
       return Front==-1 && Rear==-1;
    }
    boolean isFull()
    {
       return (Rear+1)%MAX==Front;
    }
    
    void enqueue(int data)
    {
        try
        {
            if(isEmpty())
             {
                Front=Rear=0;
             }
                queue[Rear]=data;
                Rear=Rear+1;
        }
        catch(ArrayIndexOutOfBoundsException E)
        {
            System.out.println("Queue is Full!!");
        }
        
    }
    
    void dequeue()
    {
        if(isEmpty())
        {
            System.out.println("Queue is Empty, Nothing To Dequeue");
            return;
        }
       else if(Front==Rear)
        {
            Front=Rear=-1;
        }
        else
            Front++;
    }
    
    void print()
    {
        for (int i=Front;i<Rear;i++)
        { 
            System.out.println(queue[i]);
        }
    }
}

public class Queue {

    static Scanner sc=new Scanner(System.in);
    static Queue_Operate queue;
    public static void main(String[] args) {
        queue=new Queue_Operate();
        while(true)
        {
            System.out.println("\nSelect your Operation ? : \n1.EnQueue  \n2.Dequeue \n3.Print");
            Byte ch=sc.nextByte();
            operation(ch);
        }
    }
   static void operation(Byte ch)
    {
        switch(ch)
        {
            case 1: 
                queue.enqueue(sc.nextInt());
                break;
            case 2:
                queue.dequeue();
                break;
            case 3:
                queue.print();
                break;
            default:
                System.out.println("Please Select a valid  Number\n");
                break;
  
        }
    }
    
}
